packages/plugin-ext/src/plugin/workspace.ts

```diff
@@ -349,11 +349,13 @@ export class WorkspaceExtImpl implements WorkspaceExt {
 
     updateWorkspaceFolders(start: number, deleteCount: number, ...workspaceFoldersToAdd: { uri: theia.Uri, name?: string }[]): boolean {
         const rootsToAdd = new Set<string>();
+        const rootsNameMap = new Map<string, string | undefined>();
```

Comment:
`rootsNameMap` makes `rootsToAdd` obsolete. Please use only one collection for the distinct root URIs.